lib/Raven/Serializer.php

```diff
@@ -109,6 +109,12 @@ protected function serializeValue($value)
         if (is_null($value) || is_bool($value) || is_float($value) || is_integer($value)) {
             return $value;
         } elseif (is_object($value) || gettype($value) == 'object') {
+            // If the value is a callable and expects no parameters, we evaluate the
+            // callable value by calling it.
+            if (is_callable($value) && Raven_Util::getCallableParamNum($value) === 0) {
+                return $value();
```

Comment:
The logic seems that way because callables are in essence objects, which are a stem of the serializer `ifelse` logic. If you pass in an object that is callable and can be converted to a `ReflectionFunction` only then will the callable will be evaluated.

I made a crude replica of the serialization logic here: https://3v4l.org/cZMpX

As you see only the "regular" callable is serialized by calling it. The class instances are serialized as they always have been.

See the `RavenUtil::getCallableParamNum` method that forces the callable to be an actual callable and a sole function callable (ie. an object of type `Closure`), not a class instance (which also lead me to an issue regarding PHP5+ as it doesn't have Throwables to catch which leads to fatal in my PR implementation, I should alter the check to see if the thing is a `Closure` first).

I could go and make the logic clearer in the code. As it stands the only things that are serializes/evaluated are callables that are of type `Closure`. This means no classes or instance methods will be called when serializing. I will add some tests to ensure this logic works.